storops/unity/resource/lun.py

```diff
@@ -410,6 +412,73 @@ def snapshots(self):
+                  remote_system=None):
+        """
+        Creates a replication session with a existing lun as destination.
+
+        :param dst_lun_id: destination lun id.
+        :param max_time_out_of_sync: maximum time to wait before syncing the
+            source and destination. Value `-1` means the automatic sync is not
+            performed. `0` means it is a sync replication.
+        :param replication_name: replication name.
+        :param replicate_existing_snaps: whether to replicate existing snaps.
+        :param remote_system: `UnityRemoteSystem` object. The remote system to
+            which the replication is being configured. When not specified, it
+            defaults to local system.
+        :return: created replication session.
+        """
+
+        return UnityReplicationSession.create(
+            self._cli, self.get_id(), dst_lun_id, max_time_out_of_sync,
+            name=replication_name,
+            replicate_existing_snaps=replicate_existing_snaps,
+            remote_system=remote_system)
+
+    def replicate_with_dst_resource_provisioning(self, max_time_out_of_sync,
+                                                 dst_pool_id,
+                                                 dst_lun_name=None,
+                                                 remote_system=None,
+                                                 replication_name=None,
+                                                 dst_size=None, dst_sp=None,
+                                                 is_dst_thin=None,
+                                                 dst_tiering_policy=None,
+                                                 is_dst_compression=None):
+        """
+        Creates a replication session with destination lun provisioning.
+
+        :param max_time_out_of_sync: maximum time to wait before syncing the
+            source and destination. Value `-1` means the automatic sync is not
+            performed. `0` means it is a sync replication.
+        :param dst_pool_id: id of pool to allocate destination lun.
+        :param dst_lun_name: destination lun name.
+        :param remote_system: `UnityRemoteSystem` object. The remote system to
+            which the replication is being configured. When not specified, it
+            defaults to local system.
+        :param replication_name: replication name.
+        :param dst_size: destination lun size.
+        :param dst_sp: `NodeEnum` value. Default storage processor of
+            destination lun.
+        :param is_dst_thin: indicates whether destination lun is thin or not.
+        :param dst_tiering_policy: `TieringPolicyEnum` value. Tiering policy of
+            destination lun.
+        :param is_dst_compression: indicates whether destination lun is
+            compression enabled or not.
+        :return: created replication session.
+        """
+
+        dst_size = self.size_total if dst_size is None else dst_size
```

Comment:
how about:
dst_size = dst_size or self.size_total

Reply:
Good catch. 
But should notice the case when `dst_size = 0`, `dst_size or self.size_total` returns unexpected result.
Any way, `dst_size` cannot be `0` here.

I will update the codes in next commit.